src/components/Loader/Loader.jsx

```diff
@@ -10,42 +12,82 @@ const Loader = forwardRef(
       svgClassName,
       className,
       size,
+      color,
+      type,
       id
     },
     ref
   ) => {
     const overrideClassName = backwardCompatibilityForProperties([className, svgClassName], "");
 
-    const style = useMemo(() => {
+    const sizeStyle = useMemo(() => {
       if (typeof size === "number") {
         return { width: size, height: size };
       }
       return undefined;
     }, [size]);
 
     return (
-      <div className="monday-loader-component" ref={ref} role="alert" title="loading" style={style} id={id}>
-        <svg className={`circle-loader-spinner ${overrideClassName}`} viewBox="0 0 50 50">
-          <circle className="circle-loader-spinner-path" cx="25" cy="25" r="20" fill="none" strokeWidth="5" />
+      <div
+        className={cx(styles.mondayLoaderComponent, "monday-loader-component")}
+        ref={ref}
+        role="alert"
+        title="loading"
+        style={sizeStyle}
+        id={id}
+      >
+        <svg
+          className={cx(styles.circleLoaderSpinner, overrideClassName, "circle-loader-spinner")}
+          viewBox="0 0 50 50"
+          color={color}
+        >
+          {type === Loader.types.WITH_BACKGROUND && (
+            <circle
+              className={styles.circleLoaderSpinnerBackground}
+              cx="25"
+              cy="25"
+              r="20"
+              fill="none"
+              strokeWidth="5"
+            />
+          )}
+          <circle
+            className={cx(styles.circleLoaderSpinnerPath, "circle-loader-spinner-path")}
+            cx="25"
+            cy="25"
+            r="20"
+            fill="none"
+            strokeWidth="5"
+          />
         </svg>
       </div>
     );
   }
 );
 
+Loader.colors = LOADER_COLORS;
+Loader.sizes = LOADER_SIZES;
+Loader.types = LOADER_TYPES;
+
 Loader.propTypes = {
   id: PropTypes.string,
   className: PropTypes.string,
-  /**
-   * The containing box of the loader
-   */
-  size: PropTypes.number
+  size: PropTypes.oneOf([Loader.sizes.XS, Loader.sizes.SMALL, Loader.sizes.MEDIUM, Loader.sizes.LARGE]),
+  color: PropTypes.oneOf([
+    Loader.colors.PRIMARY,
+    Loader.colors.ON_PRIMARY,
+    Loader.colors.SECONDARY,
+    Loader.colors.DARK
+  ]),
+  type: PropTypes.oneOf([Loader.types.CASUAL, Loader.types.WITH_BACKGROUND])
```

Comment:
I don't think it's a type, it sound to me it will be better to have it as boolean prop.
in the future we will maybe want to add not circle loader and then there will be 2 loader types

Reply:
also the name casual it's a bit not standard (never seen it before)